cpp/src/strings/attributes.cu

```diff
@@ -111,22 +111,24 @@ std::unique_ptr<column> counts_fn(strings_column_view const& strings,
 __global__ void count_characters_parallel_fn(column_device_view const d_strings,
                                              size_type* d_lengths)
 {
-  size_type const idx = static_cast<size_type>(threadIdx.x + blockIdx.x * blockDim.x);
-  using warp_reduce   = cub::WarpReduce<size_type>;
+  auto const idx =
+    static_cast<thread_index_type>(threadIdx.x) +
+    static_cast<thread_index_type>(blockIdx.x) * static_cast<thread_index_type>(blockDim.x);
```

Comment:
Why not `grid_1d::global_thread_id()` (same with other places with this pattern)?